src/classes/RD2_NamingService.cls

```diff
@@ -116,6 +116,27 @@ public with sharing class RD2_NamingService {
         autogenerateNames(changedRds);
     }
 
+    /***
+    * @description AutoPopulates account on Recurring Donations if the account is null and the contact is populated
+    * or the contact have been changed
+    * @param rds New/changed Recurring Donations
+    * @param oldRds Old Recurring Donations
+    */
+    public void autoPopulateAccountsOnChange(List<npe03__Recurring_Donation__c> rds, List<npe03__Recurring_Donation__c> oldRds) {
```

Comment:
Account on RD should not be populated in the naming service which only should be setting the name. This service is executed in both legacy and enhanced RD context on the insert, and in enhanced RD context on update. The RD2 trigger should be setting the donor details. It can be an inner class in the `RD2_RecurringDonations_TDTM` if the functionality cannot be created in an existing service class.